### DISCUSSION TITLE
Selected elements get lost when using scroll bars

### DISCUSSION THREAD

**Discussion by aurelio-pennacchio** [eclipse-glsp/glsp#701] [Q&A] [answered]
I activated the GLSPProjectionView in the di-config.ts file using the following line of code
`configureModelElement(context, DefaultTypes.GRAPH, GLSPGraph, GLSPProjectionView);`
Now when I have an active selection of a node of the GGraph and then I use the scrollBars I lose the selection while I was expecting to mantain the previously selected elements.
Is it a configuration issue from my side or something else?

Thank you for all your work!

**Comment by tortmayr** [comment, marked as answer]
Hi @aurelio-pennacchio ,
Unfortunately this is the default behavior of GLSP/sprotty. If you click on another UI element the currently active selection is reset. (This also happens if you don't use the scrollbar and move the viewport via dragging/panning).

The projection view is a [sprotty](https://github.com/eclipse/sprotty) feature so maybe you should also raise your question there.
My guess is that you probably have to customize the [SelectMouseListener](https://github.com/eclipse/sprotty/blob/c7368c500cd2498757a1796a98afb2e94129bb36/packages/sprotty/src/features/select/select.ts#L205) if you want to keep the selection.

Also note that there is currently an open bug that prevents keylisteners form working when using the projection view (https://github.com/eclipse/sprotty/issues/299). This is also the reason why we currently don't use it as default view. But hopefully this issue will be fixed soon.

**Reply by aurelio-pennacchio** [reply]
Thank you for all your assistance!
